content/docs/how-we-work/coding-principles-standards.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Coding Principles and Standards
+weight: 4
+description: At Axelerant, we write code not only to solve a specific problem but also to provide a reusable solution that can be used in a community level. We have principles that can guide us to the desired code standards.
+---
+
+# Coding Principles and Standards
+
+At Axelerant, we write code not only to solve a specific problem but also to provide a reusable solution that can be used in a community level. We have principles that can guide us to the desired code standards.
+
+## Coding Principles
+
+- We use the correct layer of abstraction.
+
+- We write clean, simple code and explain our purpose with comments.
+
+- We contribute back everything we can, because it supports us, our clients and the community.
+
+- We manage patches in a standard way so that we can know if code/contrib has been changed, how and why. Every change should be contributed back, and the patch name should include the drupal.org issue NID.
+
+- We write secure code.
+
+- We write code that anticipates change, so that we can implement (most) changes in an easy and Agile way.
+
+- We write API layer that doesn't assume a particular UI, so that the API could be used in multiple (and unpredictable) situations (For example, used in these UIs: Web site, drush, rest, SMS).
+
+- We remove dead code, because it's dead (and because it's confusing to leave it).
+
+- We treat one as a special case of many, so that the logic is in one place and we don't have to maintain it in two places.
+
+- When writing getter functions, consider handling the logic in the many only. A `get_one` can call the `get_many`.
```

Comment:
@Yemaneberhan-Lemma, I feel this is too specific. We don't come across needs like this often and sometimes, it is better to have a different implementation in get_one. That makes this very very subjective. What is your motivation in including this?

Reply:
I thought the content can be a good guideline. I didn't check the content is up-to-date.